zbus/Cargo.toml

```diff
@@ -133,6 +133,7 @@ tracing-subscriber = { version = "0.3.18", features = [
   "ansi",
 ], default-features = false }
 tempfile = "3.10.1"
+smol = "2.0"
```

Comment:
What's this for?

Reply:
We already have runtime abstractions that are already in use by tests. Spawning is part of them.

Reply:
But you need the executor, which is part of Connection.

Reply:
So? What's the issue?

Reply:
You can always add API to the abstraction. In any case, we definitely don't need another runtime, when we've one guaranteed to be available.

Reply:
There is no connection before spawning the task. smol is only added as dev-dependency. I welcome alternative solutions.

Reply:
> There is no connection before spawning the task

As I wrote:

> You can always add API to the abstraction. 

I just checked. There is `Executor::new` that's currently `pub(crate)`. Just make it `pub` and `#[doc(hidden)]`.

Reply:
> smol is only added as dev-dependency.

I don't mind adding the dev-dep but only if there is a justification to add it. There isn't any AFAICT.

Reply:
There is something inelegant about using block_on and a custom executor to run some standard async code test.

Reply:
> a custom executor

? It's just a think wrapper.